R/tm_t_crosstable.R

```diff
@@ -169,7 +169,20 @@ srv_t_crosstable <- function(id, data, reporter, filter_panel_api, label, x, y,
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract = list(x = x, y = y), datasets = data)
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = list(x = x, y = y),
+      datasets = data,
+      select_validation_rule = list(
+        x = shinyvalidate::sv_required("Please define column for row variable."),
+        y = shinyvalidate::sv_required("Please define column for column variable.")
+      )
+    )
+
+    iv_r <- reactive({
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule("join_fun", shinyvalidate::sv_required("Please select a joining function."))
```

Comment:
This rule should only be added if x and y are from different datasets - in the example above, if you start here:

![image](https://user-images.githubusercontent.com/15201933/210359094-0d56e97d-8ee8-4081-94b3-e88e1ed0af4c.png)
![image](https://user-images.githubusercontent.com/15201933/210359182-a10d2f39-0e41-4209-a3bf-6fd7972e40f9.png)

Then switch back to D1 without fixing the join function:
![image](https://user-images.githubusercontent.com/15201933/210359239-25057112-577e-4bdd-a6e4-9bfb1f626b02.png)

Reply:
I think I fixed it.

Reply:
you missed a "!" but now it's fixed